### DISCUSSION TITLE
Level 8: Class App\Models\[MODEL] uses generic trait Illuminate\Database\Eloquent\Factories\HasFactory but does not specify its types: TFactory

### DISCUSSION THREAD

**Discussion by nexxai** [larastan/larastan#1992] [Q&A] [answered]
I've never had to add a type to a trait before, and after some fruitless Googling, I'm hoping someone can show me the syntax for assigning a type to a trait.

**Comment by calebdw** [comment, marked as answer]
You need to use `@use`, I show an example in the PR (https://github.com/laravel/framework/pull/52005)

**Reply by nexxai** [reply]
Ahhh, perfect.  I scoured the larastan repo to see if I could find any matching PRs; should have checked the Framework repo.  Thanks for confirming!
